app/page.tsx

```diff
@@ -1,12 +1,22 @@
 import { headers } from "next/headers";
 import { ClientComponentThatConsumesPromise } from "./cc";
+import { Suspense } from "react";
 
 export default function Home() {
-	const promise = headers();
+
 
 	return (
 		<>
-			<ClientComponentThatConsumesPromise promise={promise} />
+		<h1>Hello world</h1>
+		<Suspense>
+			<DynamicArea />
+		</Suspense>
 		</>
 	);
 }
+
+function DynamicArea() {
+	const promise = headers()
+
+	return <ClientComponentThatConsumesPromise promise={promise} />
```

Comment:
worth pointing out that you don't want to send all the headers down to a client component as it exposes http-only cookies in the html. This is just a demo to show a dynamic component doesn't need to opt entire page into dynamic rendering